sigstore/_internal/trust.py

```diff
@@ -323,28 +333,39 @@ def __init__(self, inner: _SigningConfig):
         @api private
         """
         self._inner = inner
-        self._verify()
-
-    def _verify(self) -> None:
-        """
-        Performs various feats of heroism to ensure that the signing config
-        is well-formed.
-        """
 
         # must have a recognized media type.
         try:
             SigningConfig.SigningConfigType(self._inner.media_type)
         except ValueError:
             raise Error(f"unsupported signing config format: {self._inner.media_type}")
 
-        # currently not supporting other select modes
-        # TODO: Support other modes ensuring tsa_urls() and tlog_urls() work
-        if self._inner.rekor_tlog_config.selector != ServiceSelector.ANY:
-            raise Error(
-                f"unsupported tlog selector {self._inner.rekor_tlog_config.selector}"
-            )
-        if self._inner.tsa_config.selector != ServiceSelector.ANY:
-            raise Error(f"unsupported TSA selector {self._inner.tsa_config.selector}")
+        # Create lists of service protos that are valid & supported by this client
+        # Limit the TSA and tlog lists using the service selector config
+        self._tlogs = self._get_valid_services(
```

Comment:
I think it would be safer to not yet set the instance variable. Wait until we decide on a on the set of Services.

Reply:
I'll need to add yet another else-branch in that case but I suppose it is more readable

Reply:
this got resolved differently by following your later suggestion